src/plugins/dashboard/public/application/utils/use/use_dashboard_container.tsx

```diff
@@ -189,6 +225,41 @@ const createDashboardEmbeddable = async (
             ) : null;
           };
 
+          dashboardContainer.getChangesFromAppStateForContainerState = (currentContainer: any) => {
+            const appStateDashboardInput = getDashboardInput();
+            if (!dashboardContainer || isErrorEmbeddable(dashboardContainer)) {
+              return appStateDashboardInput;
+            }
+
+            const containerInput = currentContainer.getInput();
+            const differences: Partial<DashboardContainerInput> = {};
+
+            // Filters shouldn't  be compared using regular isEqual
+            if (
+              !opensearchFilters.compareFilters(
+                containerInput.filters,
+                appStateDashboardInput.filters,
+                opensearchFilters.COMPARE_ALL_OPTIONS
+              )
+            ) {
+              differences.filters = appStateDashboardInput.filters;
+            }
+
+            Object.keys(_.omit(containerInput, ['filters'])).forEach((key) => {
+              const containerValue = (containerInput as { [key: string]: unknown })[key];
+              const appStateValue = ((appStateDashboardInput as unknown) as {
+                [key: string]: unknown;
+              })[key];
+              if (!_.isEqual(containerValue, appStateValue)) {
+                (differences as { [key: string]: unknown })[key] = appStateValue;
+              }
+            });
+
+            // cloneDeep hack is needed, as there are multiple place, where container's input mutated,
+            // but values from appStateValue are deeply frozen, as they can't be mutated directly
+            return Object.values(differences).length === 0 ? undefined : _.cloneDeep(differences);
```

Comment:
Can we use `structuredClone` here?

Reply:
cloneDeep is used 94 times in this repo, while structuredClone is used 0 times. Just an observation.